driver/bpf/fillers.h

```diff
@@ -2239,6 +2251,33 @@ static __always_inline bool get_exe_writable(struct task_struct *task)
 	return false;
 }
 
+static __always_inline bool get_exe_upper_layer(struct inode *inode)
+{
+	struct super_block *sb = _READ(inode->i_sb);
+	if(sb == NULL)
+	{
+		return false;
+	}
+
+	unsigned long sb_magic = _READ(sb->s_magic);
+	if(sb_magic == PPM_OVERLAYFS_SUPER_MAGIC)
+	{
+		struct dentry *upper_dentry = NULL;
+		struct inode *lower_inode = NULL;
+		char *vfs_inode = (char *)inode;
+		
+		bpf_probe_read(&upper_dentry, sizeof(upper_dentry), vfs_inode + sizeof(struct inode));
+		bpf_probe_read(&lower_inode, sizeof(lower_inode), vfs_inode + sizeof(struct inode) + sizeof(struct dentry *));
+
+		if(!lower_inode && upper_dentry)
```

Comment:
Curiosity question: This double check condition is not redundant right? Asking because we don't seem to care about the actual fields and rather only use them to set the boolean value if there is an `upper_dentry`. Is it there is only ever either or?

 `bpf_fd_to_socktuple` seems to do similar pointer arithmetic.